src/Bicep.Core/Parsing/ParamsParser.cs

```diff
@@ -56,9 +57,9 @@ protected override SyntaxBase Declaration(params string[] expectedKeywords) =>
                     {
                         TokenType.Identifier => ValidateKeyword(current.Text) switch
                         {
-                            LanguageConstants.UsingKeyword => this.UsingDeclaration(),
-                            LanguageConstants.ExtendsKeyword => this.ExtendsDeclaration(),
-                            LanguageConstants.ParameterKeyword => this.ParameterAssignment(),
+                            LanguageConstants.UsingKeyword => this.UsingDeclaration(leadingNodes),
+                            LanguageConstants.ExtendsKeyword => this.ExtendsDeclaration(leadingNodes),
+                            LanguageConstants.ParameterKeyword => this.ParameterAssignment(leadingNodes),
```

Comment:
This isn't the right approach. It's not the job of the parser to reject decorators. Follow the code for the following validation error to get an understanding of how this is handled elsewhere:

```
The name "foo" does not exist in the current context.bicep[BCP057](https://aka.ms/bicep/core-diagnostics#BCP057
```